frontend/frontend/src/screens/ChartScreen.js

```diff
@@ -0,0 +1,210 @@
+
+const ChartScreen = ({ history }) => {
+    const navigate = useNavigate()
+
+    const [error, setError] = useState('')
+    const [loading, setLoading] = useState(false)
+    const [expenses, setExpenses] = useState([])
+
+    const [errorCategories, setErrorCategories] = useState('')
+    const [loadingCategories, setLoadingCategories] = useState(false)
+    const [categories, setCategories] = useState([])
+
+    const [month, setMonth] = useState('august')
+    const months = [
+        'january',
+        'february',
+        'mars',
+        'april',
+        'may',
+        'june',
+        'july',
+        'august',
+        'september',
+        'october',
+        'november',
+        'december',
+    ]
+    const getExpenses = async () => {
+        try {
+            setLoading(true)
+            const config = {
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            }
+            const { data } = await axios.get(`http://localhost:5000/expenses`, config)
+
+            setExpenses(data)
+            console.log('expenses', expenses.length)
+        } catch (errors) {
+            setError(
+                errors.response && errors.response.data.detail
+                    ? errors.response.data.detail
+                    : errors.message
+            )
+        } finally {
+            setLoading(false)
+        }
+    }
+
+    const getCategories = async () => {
+        try {
+            setLoadingCategories(true)
+            const config = {
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            }
+            const { data } = await axios.get(`http://localhost:5000/categories`, config)
+
+            setCategories(data)
+        } catch (errors) {
+            setErrorCategories(
+                errors.response && errors.response.data.detail
+                    ? errors.response.data.detail
+                    : errors.message
+            )
+        } finally {
+            setLoadingCategories(false)
+        }
+    }
+    function random_rgba() {
+        var o = Math.round,
+            r = Math.random,
+            s = 255
+        return (
+            'rgba(' + o(r() * s) + ',' + o(r() * s) + ',' + o(r() * s) + ',' + r().toFixed(1) + ')'
+        )
+    }
+    const generate_chart_data = () => {
+        var categors = []
+        var spendings = []
+
+        categories.map((category) => {
+            categors.push(category.title)
+            spendings.push(
+                expenses.reduce((acc, expense) => {
+                    if (
+                        expense.category == category._id &&
+                        months[Number(expense.date.substring(5, 7))] === month
+                    ) {
+                        return acc + expense.amount
+                    } else {
+                        return acc
+                    }
+                }, 0)
+            )
+        })
```

Comment:
Too complex

Reply:
This part is related to the table that can be removed because it is not a requirement

Reply:
Are you sure you use correct iteration method here?

Reply:
Do you mean that the results are incorrect or that the iteration method is not suitable (efficient)?